daml-lf/spec/daml-lf-1.rst

```diff
@@ -2889,6 +2920,17 @@ as described by the ledger model::
        ⇓ᵤ
      Ok ('Some' @('ContractId' Mod:T) cid, ε) ‖ (st; keys)
 
+     'tpl' (x : T) ↦ { …, 'key' @σ eₖ eₘ }  ∈ 〚Ξ〛Mod
+     'fetch_by_key' @Mod:T vₖ ‖ (st; keys)  ⇓ᵤ  Err t
+   —————————————————————————————————————————————————————————————————————— EvUpdExercByKeyFetchErr
+     'exercise_by_key' Mod:T.Ch vₖ v₁ ‖ (st; keys)  ⇓ᵤ  Err t
+
+     'tpl' (x : T) ↦ { …, 'key' @σ eₖ eₘ }  ∈ 〚Ξ〛Mod
+     'lookup_by_key' @Mod:T vₖ ‖ (st; keys)  ⇓ᵤ  Ok ⟨'contractId': cid, 'contract': vₜ⟩ ‖ (st'; keys')
+     'exercise_without_actor' Mod:T.Ch cid v₁ ‖ (st'; keys')  ⇓ᵤ  ur
+   —————————————————————————————————————————————————————————————————————— EvUpdExercByKeyExercise
+     'exercise_by_key' Mod:T.Ch vₖ v₁ ‖ (st; keys)  ⇓ᵤ  ur
```

Comment:
This is a nice approach, but the `ur` includes the ledger transaction, which means we aren't distinguishing between "exercise" and "exercise by key". Maybe the semantics needs to be reformulated anyway to track the transaction structure more closely, so I'm ok leaving it like this for now.

Reply:
You are right, this does not make the distinction between "exercise" and "exercise by key". But currently the implementation does not neither, so it is probably fine to let the semantics like that until we introduce a new transaction format.

Reply:
Let's leave it as is for now and fix it when we've introduced the new transaction format.